### DISCUSSION TITLE
Fields order in composite primary key

### DISCUSSION THREAD

**Discussion by heahoh** [doctrine/doctrine2#10761] [Support Questions]
Hi!

Got an unexpected behavior (not documented as i see) when creating composite primary key

I has two entity with OneToMany relation - order and order.item
Both entity has external identity - order has unique digital number, item - non-unique SKU (but in order context it's unique)
`
Order
  @Id
  private int externalId
  
  @OneToMany
  private Collection items
`

`
Item
  @Id
  private string sku
  
  @Id
  private Order order
`

For selectivity reasons I want to create primary_key like (order_id, sku) but `doctrine:schema:update --dump-sql` creates this one - `primary_key(sku, order_id)`
If I swap lines of `order` and `sku` fields declaration to this

`
Item
  @Id
  private Order order
  
  @Id
  private string sku
`
Than I get expected declaration of primary_key

Seems like sort of UB
Codestyle fixes can leads to changing table primary key and behavior not documented in documentation

**Comment by derrabus** [comment]
I don't really see why it's unexpected that ID fields are taken into account in the very order that you define them in.
